client/mixnet/src/config.rs

```diff
@@ -0,0 +1,90 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2022-2023 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+use std::time::Duration;
+
+/// Substrate-specific mixnet configuration.
+#[derive(Clone, Debug)]
+pub struct SubstrateConfig {
```

Comment:
I agree that the name is not great, but I don't have any better ideas. I named this `SubstrateConfig` because the other parts of the configuration are defined in the `mixnet` crate and are thus not specific to Substrate.

Reply:
Think I'll keep it as-is. The other config structs are handed off to the relevant subsystems in `run`. References to the `SubstrateConfig` are passed around within `sc-mixnet`. Could pass around `Config` references instead, but that would require extra cloning and pointlessly give things access to `CoreConfig` etc.